packages/twenty-server/src/integrations/message-queue/drivers/bullmq.driver.ts

```diff
@@ -49,20 +49,59 @@ export class BullMQDriver implements MessageQueueDriver {
     this.workerMap[queueName] = worker;
   }
 
-  async add<T>(
+  async schedule<T>(
     queueName: MessageQueue,
     jobName: string,
     data: T,
+    pattern: string,
     options?: QueueJobOptions,
   ): Promise<void> {
     if (!this.queueMap[queueName]) {
       throw new Error(
         `Queue ${queueName} is not registered, make sure you have added it as a queue provider`,
       );
     }
-    await this.queueMap[queueName].add(jobName, data, {
-      jobId: options?.id,
-      priority: options?.priority,
+    const queueOptions = Object.assign(
+      {},
+      options?.id ? { jobId: options.id } : null,
+      options?.priority ? { priority: options.priority } : null,
+      {
+        repeat: {
+          pattern,
+        },
+      },
+    );
+
+    await this.queueMap[queueName].add(jobName, data, queueOptions);
+  }
+
+  async unschedule(
+    queueName: MessageQueue,
+    jobName: string,
+    pattern: string,
+  ): Promise<void> {
+    await this.queueMap[queueName].removeRepeatable(jobName, {
+      pattern,
     });
   }
+
+  async add<T>(
+    queueName: MessageQueue,
+    jobName: string,
+    data: T,
+    options?: QueueJobOptions,
+  ): Promise<void> {
+    if (!this.queueMap[queueName]) {
+      throw new Error(
+        `Queue ${queueName} is not registered, make sure you have added it as a queue provider`,
+      );
+    }
+    const queueOptions = Object.assign(
```

Comment:
I'm not fully sure why you need this?

Reply:
need that to add `key: values` only if values exists. Maybe a better way to do it exists